_source/_posts/2022-05-19-angular-microfrontend-deploy.md

```diff
@@ -0,0 +1,292 @@
+
+Next, add the deploy URI as a Trusted Origin in Okta. Navigate to **Security** > **API** and then navigate to the **Trusted Origins** tab. Press **+ Add Origin**. Add the deploy location and select **CORS** for the type.
+
+{% img blog/angular-microfrontend-deploy/okta-configure-trusted-origin.jpg alt:"Okta configuration for adding the Vercel deploy URI of shell application to trusted origin. Type 'CORS' is selected." width:"800" %}{: .center-image }
+
+Commit your changes and push to your main branch. After the build succeeds, you should be able to use the deployed site from end to end!
+
+
+## Secure your micro frontends
+
+Right now, you can access what should be protected routes by manually typing in the full URI to the profile route. Even though you don't see any profile information, we aren't guarding the route. Let's make the site a little more secure by protecting the route.
+
+Open `projects/shell/src/app/app-routing.module.ts`. We'll add a route guard that comes out of the box with Okta's Angular SDK. Update the route definition for the `profile` route as shown below.
+
+```ts
+import { OktaCallbackComponent, OktaAuthGuard} from '@okta/okta-angular';
+
+{
+  path: 'profile',
+  loadChildren: () => loadRemoteModule({
+    type: 'module',
+    remoteEntry: `${environment.mfe.mfeProfile}/remoteEntry.js`,
+    exposedModule: './Module'
+  }).then(m => m.ProfileModule),
+  canActivate: [OktaAuthGuard]
+},
+```
+
+Now if you navigate directly to the profile route, you'll first redirect to Okta's sign-in page, then redirect back to the profile route. 
+
+Route protection works well from within the micro-frontend shell. But what if you navigate directly to the URI where the profile micro frontend resides? You will still be able to navigate to the profile route within the micro frontend because there isn't a guard within the profile application. 
+
+This brings us to an interesting concept. In order to fully secure your micro frontends, you should protect routes defined within your micro frontend as well. To add the Okta route guard, you'll need to import the `OktaAuthModule` into the `mfe-profile` application's `AppModule`, and add the same configuration as you did for the `shell` application. The Module Federation configuration shares the Okta library between the applications when accessed via the host application. In contrast, `mfe-profile` will need its own instance of authenticated state when accessed in isolation from the `shell` application. Having its own instance of authenticated state means you should add sign in and sign out capability in the `AppComponent` of the `mfe-profile` application too. This will also allow you to test each micro-frontend application independently.
```

Comment:
```suggestion
This brings us to an interesting concept. In order to fully secure your micro frontends, you should protect routes defined within your micro frontend as well. To add the Okta route guard, you'll need to import the `OktaAuthModule` into the `mfe-profile` application's `AppModule`, and add the same configuration as you did for the `shell` application. The Module Federation configuration shares the Okta library between the applications when accessed via the host application. In contrast, `mfe-profile` will need its own instance of authenticated state when accessed in isolation from the `shell` application. Having its own instance of authenticated state means you should add sign-in and sign-out capability in the `AppComponent` of the `mfe-profile` application too. This will also allow you to test each micro-frontend application independently.
```